tests/linalg_test.py

```diff
@@ -1619,6 +1619,38 @@ def testToeplitzConstructionWithKnownCases(self):
       [2, 1, 4, 5],
       [3, 2, 1, 4]], dtype=np.float32))
 
+  @jtu.sample_product(
+    base_shape=[1, 2, 3, 10, 100],
+    dtype=float_types + complex_types)
+  def testNullSpace(self, base_shape, dtype):
+    for shape in [
+      (2, base_shape),
+      (base_shape, 2),
+      (1 + base_shape // 2, base_shape),
+    ]:
```

Comment:
Let's try to keep iterations in the test parameterization, not in the test itself. The reason is, if any of these fails it prevents subsequent iterations from running. It's better to have all test failures reported at once.

Reply:
makes total sense will correct